src/Jetpack_Usage.php

```diff
@@ -0,0 +1,160 @@
+		add_action( 'jpc_register_fail', array( $this, 'jpc_register_fail' ) );
+		add_action( 'jpc_register_success', array( $this, 'jpc_register_success' ) );
+	}
+
+	/* Activated module */
+	public function track_activate_module( $module ) {
+		$this->tracking->record_user_event( 'module_activated', array( 'module' => $module ) );
+	}
+
+	/* Deactivated module */
+	public function track_deactivate_module( $module ) {
+		$this->tracking->record_user_event( 'module_deactivated', array( 'module' => $module ) );
+	}
+
+	/* User has linked their account */
+	public function track_user_linked() {
+		$user_id = get_current_user_id();
+		$anon_id = get_user_meta( $user_id, 'jetpack_tracks_anon_id', true );
+
+		if ( $anon_id ) {
+			$this->tracking->record_user_event( '_aliasUser', array( 'anonId' => $anon_id ) );
+			delete_user_meta( $user_id, 'jetpack_tracks_anon_id' );
+			if ( ! headers_sent() ) {
+				setcookie( 'tk_ai', 'expired', time() - 1000 );
+			}
+		}
+
+		$wpcom_user_data = self::get_connected_user_data( $user_id );
+		update_user_meta( $user_id, 'jetpack_tracks_wpcom_id', $wpcom_user_data['ID'] );
+
+		$this->tracking->record_user_event( 'wpa_user_linked', array() );
+	}
+
+	/**
+	 * Track that we've begun verifying the secrets.
+	 *
+	 * @access public
+	 *
+	 * @param string $action Type of secret (one of 'register', 'authorize', 'publicize').
+	 * @param \WP_User $user The user object.
+	 */
+	public function track_jetpack_verify_secrets_begin( $action, $user ) {
+		$this->tracking->record_user_event( "jpc_verify_{$action}_begin", array(), $user );
+	}
+
+	/**
+	 * Track that we've succeeded in verifying the secrets.
+	 *
+	 * @access public
+	 *
+	 * @param string $action Type of secret (one of 'register', 'authorize', 'publicize').
+	 * @param \WP_User $user The user object.
+	 */
+	public function track_jetpack_verify_secrets_success( $action, $user ) {
+		$this->tracking->record_user_event( "jpc_verify_{$action}_success", array(), $user );
+	}
+
+	/**
+	 * Track that we've failed verifying the secrets.
+	 *
+	 * @access public
+	 *
+	 * @param string $action Type of secret (one of 'register', 'authorize', 'publicize').
+	 * @param \WP_User $user The user object.
+	 * @param \WP_Error $error Error object.
+	 */
+	public function track_jetpack_verify_secrets_fail( $action, $user, $error ) {
+		$this->tracking->record_user_event(
+			"jpc_verify_{$action}_fail",
+			array(
+				'error_code'    => $error->get_error_code(),
+				'error_message' => $error->get_error_message(),
+			),
+			$user
+		);
+	}
+
+	/* Failed login attempts */
+	public function track_failed_login_attempts( $login ) {
+		require_once JETPACK__PLUGIN_DIR . 'modules/protect/shared-functions.php';
+		$this->tracking->record_user_event(
+			'failed_login',
+			array(
+				'origin_ip' => jetpack_protect_get_ip(),
+				'login'     => $login,
+			)
+		);
+	}
+
+	function jpc_register_fail( $error, $registered ) {
+		do_action( 'jpc_register_fail', $error );
```

Comment:
Won't this lead to a loop?

Reply:
oops, that was my copy and paste to get the name of the action  :)